src/block/Sugarcane.php

```diff
@@ -123,4 +129,28 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 
 		return false;
 	}
+
+	private function isSupportedByWater() : bool{
+		$down = $this->getSide(Facing::DOWN);
+		foreach($down->getHorizontalSides() as $sideBlock){
+			$blockId = $sideBlock->getTypeId();
+
+			if ($blockId === BlockTypeIds::WATER || $blockId === BlockTypeIds::FROSTED_ICE) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public function onNearbyBlockChange() : void {
+		$bottom = $this->seekToBottom();
+		if (!$this->position->equals($bottom)) {
```

Comment:
It would be better if you combine these 2 conditions into one (e.g `if($this->position->equals($bottom) && !$this->isSupportedByWater()){}`

Reply:
By the way, you can just check that the down side of the block doesn't have same type id instead of seeking to the bottom (like it is done in `onRandomTick()`). It would be a little bit faster in large crop farms.